auto_install/install.sh

```diff
@@ -2706,13 +2706,32 @@ confUnattendedUpgrades(){
 			fi
 		fi
 	elif [[ "${PKG_MANAGER}" == 'apk' ]]; then
-		echo 'https://dl-cdn.alpinelinux.org/alpine/edge/testing/' | ${SUDO} tee -a /etc/apk/repositories && ${SUDO} apk -q update
+		## install dependencies
+		# shellcheck disable=SC2086
+		${SUDO} ${PKG_INSTALL} unzip
 
-		PIVPN_DEPS=(apk-autoupdate)
+		## download binaeries
+		curl -fLo master.zip https://github.com/jirutka/apk-autoupdate/archive/refs/heads/master.zip
+		unzip -q master.zip
 
-		installDependentPackages PIVPN_DEPS[@]
+		cd apk-autoupdate-master || exit 1
 
-		${SUDO} sed -i -E -e '/^https:\/\/dl\-cdn\.alpinelinux\.org\/alpine\/edge\/testing\/$/d' /etc/apk/repositories && ${SUDO} apk -q update
+		## personalize binaries
+		sed -i -E -e 's/^(prefix\s*:=).*/\1 \/usr/' Makefile
+
+		## install
+		${SUDO} make install
+
+		if ! command -v apk-autoupdate &> /dev/null; then
+			echo "::: Failed to compile and install apk-autoupdate!"
+			exit 1
+		fi
+
+		cd ..
+
+		## remove useless files
+		rm master.zip
+		rm -rf apk-autoupdate-master
```

Comment:
Permission denied on `curl` because you are in the `/usr/local/src/pivpn` folder.

I recommend downloading in a randomly generated directory, for `grepcidr` too.

```
if ! down_dir=$(mktemp -d); then
	echo "::: Failed to create download directory for apk-autoupdate!"
	exit 1
fi
```

...before `curl`, and cd to `$down_dir` or specify absolute path.